inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
+#define INTERLOCKEDIFIF_H
+
+#include <stdint.h>
+
+long InterlockedIFAdd( 
```

Comment:
>InterlockedIFAdd [](start = 5, length = 16)

should have real_blablalba - 1x.c file, 2x.h files. Just in case we never want to use the mocked function